src/cargo/core/compiler/compilation.rs

```diff
@@ -85,7 +85,7 @@ impl<'cfg> Compilation<'cfg> {
     ) -> CargoResult<Compilation<'cfg>> {
         let mut rustc = bcx.rustc.process();
 
-        let mut primary_unit_rustc_process = bcx.build_config.primary_unit_rustc.clone();
+        let mut primary_unit_rustc_process = bcx.rustc.workspace_process();
```

Comment:
I would leave this similar to what it was before, so that `cargo fix` can override the rustc used. Maybe something like:

```rust
        let mut primary_unit_rustc_process = bcx
            .build_config
            .primary_unit_rustc
            .clone()
            .or_else(|| bcx.rustc.workspace_process());
```

Reply:
I'm worried about the lack of symmetry with this design.

If we're going with "workspace" for clippy and "primary" for fix then doing it this way mixes the usage of the workspace process with the primary wrapper. Hopefully the primary packages are strictly a subset of the workspace ones so this might be fine but I feel like by keeping the filtering of "apply fixes for these" separate from "use this on all workspace crates" is just a bit less complex / intermixed.